core/src/modules/ops/destination.ts

```diff
@@ -981,6 +1010,18 @@ export namespace DestinationOps {
       await app.checkAndUpdateParallelism("decr");
     }
 
+    for (const _export of _exports) {
+      // release locks for all of the exports that were processed regardless of outcome - re-enqueues are handled elsewhere
+      const { releaseLock } = await getLock(
+        `${_export.recordId}:${_export.destinationId}`,
+        _export.id,
+        null,
+        "export"
+      );
```

Comment:
Ah, I see while you are already returning `releaseLock` now.  Perhaps above in the first instance of getLock, you could store all the `releaseLock`s you get in an array and `Promise.all(locks.map(releaseLock => releaseLock())`.  This way, you don't need to talk to redis again

Reply:
It's also probably a good idea to wrap most of this method in a try/finally block so you can still release the locks if something goes wrong

Reply:
I've done that with locks... implementing the try/finally block sounds right but the scoping of it is escaping me.  There are a handful of loops happening here, any of which could fail.